keel-redis/src/test/kotlin/com/netflix/spinnaker/keel/redis/RedisLockTests.kt

```diff
@@ -0,0 +1,36 @@
+package com.netflix.spinnaker.keel.redis
+
+import com.netflix.spinnaker.keel.sync.LockTests
+import com.netflix.spinnaker.kork.jedis.EmbeddedRedis
+import com.netflix.spinnaker.kork.jedis.JedisClientDelegate
+import org.junit.jupiter.api.AfterAll
+import redis.clients.jedis.Jedis
+import java.time.Clock
+import java.time.Duration
+
+internal object RedisLockTests : LockTests<RedisLock>() {
+
+  private val embeddedRedis = EmbeddedRedis.embed()
+  private val redisClient = JedisClientDelegate(embeddedRedis.pool)
+
+  override fun flush() {
+    redisClient.withCommandsClient { redis -> (redis as Jedis).flushAll() }
+  }
+
+  override fun simulateTimePassing(name: String, duration: Duration) {
+    redisClient.withCommandsClient { redis ->
+      val ttl = redis.pttl("lock:$name") - duration.toMillis()
+      redis.pexpire("lock:$name", Math.max(0L, ttl))
```

Comment:
Hackery to decrement the TTL of a Redis key in sync with the mutable clock

Reply:
[nice](https://youtu.be/Grdev5ktOGw)